### PR TITLE
Replication slots for HA

### LINKED ISSUE #8: Replication slots for HA
Cloud Native Postgres currently relies only on the archive log to synchronize those standby servers that have fallen out of sync. However, for some high workload scenarios, this can be inefficient. We need to introduce a smart way for the operator to manage replication slots on the primary and designated primary instances, so that the operator itself regularly moves forward the WAL location based on the most delayed standby. The operator must also remove the replication slot from the former primary and add it to the new one following a failover or switchover.

### PR DESCRIPTION
**Pull request by mnencia**
Closes #8

### CODE REVIEW

**Review by armru — APPROVED** [review]

File: internal/cmd/manager/instance/run/cmd.go [hunk 1]
```diff
@@ -178,6 +179,12 @@ func runSubCommand(ctx context.Context, instance *postgres.Instance) error {
 		return err
 	}
 
+	slotReplicator := runner.NewReplicator(instance)
+	if err = mgr.Add(slotReplicator); err != nil {
+		setupLog.Error(err, "unable to create slot slotReplicator")
```
Comment (phisco) [inline internal/cmd/manager/instance/run/cmd.go:184]:
```suggestion
		setupLog.Error(err, "unable to create slot Replicator")
```

Comment (mnencia) [reply internal/cmd/manager/instance/run/cmd.go]: @leonardoce fixed it in his review.

**Review by gbartolini — APPROVED** [review]

**Review by leonardoce — APPROVED** [review]

**Pull request merged**